src/promptflow/promptflow/_core/tool_meta_generator.py

```diff
@@ -221,6 +232,21 @@ def collect_tool_function_in_module(m):
     return tools[0]
 
 
+def collect_script_tools(m):
+    functions = collect_tool_functions_in_module(m)
+    methods = collect_tool_methods_with_init_inputs_in_module(m)
+    num_tools = len(functions) + len(methods)
+    if num_tools == 0:
+        raise NoToolDefined("No tool found in the python script.")
```

Comment:
Please redefine the error message refer to collect_tool_function_in_module func.